### PR TITLE
moves functions from httomo #27

### PR DESCRIPTION
**Pull request by dkazanc**

### CODE REVIEW

File: src/httomolib/prep/stripe.py [hunk 1]
```diff
@@ -119,4 +91,101 @@ def _rs_sort(sinogram, size, matindex, dim):
...
+    from larix.methods.misc import STRIPES_MERGE
+    
+    gradientX = _gradient(data, 2)    
+    med_val = np.median(np.abs(gradientX).flatten(), axis=0)
+    
+    # we get a local stats here, needs to be adopted for global stats
+    mask_stripe = np.zeros_like(data,dtype="uint8")    
+    mask_stripe[data > med_val/threshold_stripes] = 1
+    
+    # merge stripes that are close to each other
```
Comment (namannimmo10) [inline src/httomolib/prep/stripe.py:236]:
```suggestion
    # merge stripes that are close to each other
    mask_stripe_merged = STRIPES_MERGE(mask_stripe, stripe_width_max_perc, mask_dilate, ncore)
    
    return mask_stripe_merged

```

File: src/httomolib/misc/images.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    comm: int
+        MPI communicator.
+    axis : int, optional
+        Specify the axis to use to slice the data (if data is 3D array).
+    file_format : str, optional
+        Specify file format, e.g. "png", "jpeg" or "tiff".
+    bits : int, optional
+            Specify the number of bits (8, 16 or 32-bit).
+    jpeg_quality : int, optional
+            Specify the quality of the jpeg image.
```
Comment (namannimmo10) [inline src/httomolib/misc/images.py]:
```suggestion
    bits : int, optional
        Specify the number of bits (8, 16, or 32-bit).
    jpeg_quality : int, optional
        Specify the quality of the jpeg image.
```

File: src/httomolib/misc/images.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    """rescale to the bit chosen and save an image
+
+    Args:
+        array2d (array): given array
+        glob_stats (tuple): statistics for normalisation
+        bits (int): chosen bits number
+        jpeg_quality (int): chosen quality for jpegs
+        path_to_out_file (str): full path to the file
+    """
+    np.nan_to_num(array2d, copy=False, nan=0.0, posinf=0, neginf=0)
```
Comment (namannimmo10) [inline src/httomolib/misc/images.py]:
```suggestion
    array2d = np.nan_to_num(array2d, copy=False, nan=0.0, posinf=0, neginf=0)
```

File: src/httomolib/misc/segm.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    """gets data binary segmented into a mask  
+    
+    """
+    if otsu:
+        # get the intensity value based on Otsu
+        val_intensity = filters.threshold_otsu(data)
+    if foreground:
+        mask[data > val_intensity] = 1
+    else:
+        mask[data <= val_intensity] = 1
```
Comment (namannimmo10) [inline src/httomolib/misc/segm.py]:
```suggestion
    mask[data > val_intensity if foreground else data <= val_intensity] = 1
```

File: tests/test_misc/test_segm.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+def test_binary_thresholding():
+    _data = np.zeros(shape=(100, 100, 100))
+    _binary_mask = binary_thresholding(_data, 0.5)
+    for _ in range(5):
+        assert np.all(_binary_mask == 0)
+
+    #: testing binary_thresholding on tomo_standard data
+    binary_mask = binary_thresholding(host_data, 0.1)
+    for _ in range(5):
+        assert np.all(binary_mask == 1)
```
Comment (namannimmo10) [inline tests/test_misc/test_segm.py:22]: @dkazanc -- can you please check if this is the expected output?

Comment (namannimmo10) [reply tests/test_misc/test_segm.py:22]: Maybe using imagej?

File: src/httomolib/misc/segm.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    Performs binary thresholding to the input data
+
+    Parameters
+    ----------
+    data : ndarray
+        Input array.
+    val_intensity: float, optional
+        The grayscale intensity value that defines the binary threshold.
+        Defaults to 0.1
+    otsu: str, optional
```
Comment (namannimmo10) [inline src/httomolib/misc/segm.py]:
```suggestion
    otsu: bool, optional
```

**Comment by namannimmo10** [comment]
Unfortunately, `detect_stripes` and `merge_stripes` (taken from the Larix toolbox) take a lot of time to execute. Down the road, we can use the CPU implementations from TomoPy (after we add them there) and write GPU implementation in httomolib.

**Comment by dkazanc** [comment]
> )

they will be accelerated a bit, but we also need to apply them to one slice only. You can leave that for now.

**Comment by namannimmo10** [comment]
Now it looks much better, still, we need to fix a lot of things, including the corner cases. Can you please check if all tests pass for you locally? Please merge if they do.

**Comment by namannimmo10** [comment]
OK, I squashed all the commits into one and merged https://github.com/DiamondLightSource/httomolib/commit/2b0ab9c9be266914dcd9313a503d08b40da5820b.
